pkg/cloudprovider/aws/fleet/capacity.go

```diff
@@ -0,0 +1,183 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleet
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+)
+
+type Capacity struct {
+	spec                   *v1alpha1.ProvisionerSpec
+	ec2                    ec2iface.EC2API
+	launchTemplateProvider *LaunchTemplateProvider
+	subnetProvider         *SubnetProvider
+	nodeFactory            *NodeFactory
+}
+
+// Create a set of nodes given the constraints
+func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.CapacityConstraints) ([]*v1.Node, error) {
+	// 1. Select a zone
+	zone, err := c.selectZone(ctx, constraints)
+	if err != nil {
+		return nil, fmt.Errorf("getting zone, %w", err)
+	}
+
+	// 2. Select a zubnet, limited to selected zone
```

Comment:
Typo: `zubnet` (although kind of cool sounding!)

Reply:
![image](https://user-images.githubusercontent.com/2585992/106530238-cb1aac80-64a0-11eb-9c8e-8dfa244e0702.png)